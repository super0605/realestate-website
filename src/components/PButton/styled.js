import styled, { css } from "styled-components";
import { Button, Typography } from "antd";

const { Text } = Typography;

const resetStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 6px 10.5px 6px 10.5px;
`;

export const CustomButton = styled(Button)`
  /* style based on the button type */
  ${({ ptype }) => ptype === "reset" && resetStyle}
  ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].textStyle}
  background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].bgColor};
  color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].color};

  /* general style */
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  padding-top: ${({ ptop }) => ptop};
  padding-bottom: ${({ pbottom }) => pbottom};
  padding-left: ${({ pleft }) => pleft};
  padding-right: ${({ pright }) => pright};
  margin-top: ${({ mtop }) => mtop};
  margin-bottom: ${({ mbottom }) => mbottom};
  margin-left: ${({ mleft }) => mleft};
  margin-right: ${({ mright }) => mright};
  border: 0px;

  &.ant-btn:hover {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].hoverBgColor}!important;
    span {
      color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].hoverColor};
    }
  }
  &.ant-btn:focus {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].bgColor};
  }
  &.ant-btn:active {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].bgColor};
  }
  &.ant-btn[disabled] {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].disabledBgColor};
    opacity: 0.4;
  }
  &.ant-btn[disabled]:hover {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].disabledBgColor};
    span {
      color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].color};
    }
  }
  &.ant-btn[disabled]:focus {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].disabledBgColor};
    span {
      color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].color};
    }
  }
  &.ant-btn[disabled]:active {
    background-color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].disabledBgColor};
    span {
      color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].color};
    }
  }
`;

export const ButtonText = styled(Text)`
  /* style based on the button type */
  ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].textStyle}
  color: ${({ theme, ptype }) => theme.components.button.buttonPrime[ptype].color};
`;
