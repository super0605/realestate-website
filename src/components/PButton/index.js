import React from "react";
import PropTypes from "prop-types";
import { CustomButton, ButtonText } from "./styled";
import { IconReset } from "../../assets/icons";
import { LoadingOutlined } from "@ant-design/icons";

const btnMainType = {
  reset: "reset",
  save: "save",
  default: "default",
  reject: "reject",
};

const PButton = ({
  ptype,
  pname,
  pspin,
  ploading,
  width,
  height,
  pleft,
  pright,
  pbottom,
  ptop,
  mleft,
  mright,
  mbottom,
  mtop,
  children,
  ...rest
}) => {
  return (
    <CustomButton
      ptype={ptype}
      height={height}
      width={width}
      pleft={pleft}
      pright={pright}
      ptop={ptop}
      pbottom={pbottom}
      mleft={mleft}
      mright={mright}
      mtop={mtop}
      mbottom={mbottom}
      {...rest}
    >
      {children ? (
        children
      ) : (
        <React.Fragment>
          {ptype === btnMainType.reset && <IconReset />}
          <ButtonText ptype={ptype}>{pname}</ButtonText>
          {pspin && ploading && <LoadingOutlined />}
        </React.Fragment>
      )}
    </CustomButton>
  );
};

PButton.propTypes = {
  ptype: PropTypes.string,
  pname: PropTypes.string,
  pspin: PropTypes.bool,
  ploading: PropTypes.bool,
  width: PropTypes.string,
  height: PropTypes.string,
  pleft: PropTypes.string,
  pright: PropTypes.string,
  ptop: PropTypes.string,
  pbottom: PropTypes.string,
  mleft: PropTypes.string,
  mright: PropTypes.string,
  mtop: PropTypes.string,
  mbottom: PropTypes.string,
  children: PropTypes.node,
};

PButton.defaultProps = {
  ptype: btnMainType.default,
  pspin: false,
  ploading: false,
  pname: "BUTTON",
  height: "38px",
};

export default PButton;
