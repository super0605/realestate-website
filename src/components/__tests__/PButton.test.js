import * as React from "react";
import { mount, shallow } from "enzyme";
import PButton from "../PButton";

describe("PButton Component", () => {
  let props;
  let mountedPButton;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedPButton) {
        mountedPButton = bShallow
        ? shallow(<PButton {...props} />)
        : mount(<PButton {...props} />);
    }
    return mountedPButton;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
