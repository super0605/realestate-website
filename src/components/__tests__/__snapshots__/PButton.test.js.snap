// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PButton Component should render without problems 1`] = `
<Styled(Button)
  height="38px"
  ptype="default"
>
  <Styled(Text)
    ptype="default"
  >
    BUTTON
  </Styled(Text)>
</Styled(Button)>
`;
